model-api-gen-runtime/src/commonMain/kotlin/org/modelix/metamodel/GeneratedConcept.kt

```diff
@@ -149,7 +149,10 @@ abstract class GeneratedConcept<NodeT : ITypedNode, ConceptT : ITypedConcept>(
     }
 
     override fun isExactly(concept: IConcept?): Boolean {
-        return concept == this
+        if (concept == null) return false
+        if (concept == this) return true
+        if (concept.getReference().getUID() == getReference().getUID()) return true
```

Comment:
**Question:** Why do we need the IConceptReference instead of just the IConcept to check the UID?

Reply:
Replaced it by getting the UID directly from the IConcept.